docs/architecture.md

```diff
@@ -54,14 +64,16 @@ def my_sub_command(entity_ctx):
     entity_ctx.destroy(href)
 ```
 
-### Version dependent code paths
+#### Version dependent code paths
 
 Each Pulp CLI release is designed to support multiple Pulp server versions and the CLI itself is versioned independently of any version of the Pulp server components.
 It is supposed to be able to communicate with different combinations of server component versions at the same time.
 Because of this, it might be necessary to guard certain features and workarounds by checking against the available server plugin version.
+
 As a rule of thumb, all necessary workarounds should be implemented in the corresponding `Context` objects to provide a consistent interface to the command callbacks.
-To facilitate diverting code paths depending on plugin versions, the `PulpContext` provides the `needs_plugin` and `has_plugin` methods.
-While `has_plugin` will evaluete immediately, `needs_plugin` can be seen as a deferred assertion.
+To facilitate diverting code paths depending on plugin versions, the `PulpContext` provides the `needs_plugin` and `has_plugin` methods, both of which accept a [`PluginRequirement`]() object to describe dependencies on server components.
```

Comment:
Is the PluginRequirement here supposed to be a link? Do we need to define it, or is there some magic to hook it up automatically?

Reply:
I've missed that, we have to define.
Later I'll do some research on linters for documentation, they do a better job than us for certain things,